### PR TITLE
Update react native 0.50.3

### LINKED ISSUE #1690: Upgrade React Native from 0.43 to latest
### User Story

As a developer, I want to run the latest version of React Native so that I can use libraries that depend on it.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: We are currently using 0.43. 0.47 is out and multiple libraries (`react-native-fcm` and `react-native-splashscreen`, for two examples) depend on the latest version on React Native. Keeping up to date allows us to rule out classes of bugs, use recent documentation, as well as get recent bugfixes and enhancements.

### PR DESCRIPTION
**Pull request by yenda**
fixes #1690

```
TODOS:

- [x] Android migration
- [x] iOS migration
- [x] move all github repos in package.json to Status repos
```
Test:
This changes react-native version and update most libraries in the process. But there is no changes to our code itself so if the app loads, tabs work, scrolling work, drawer works it should be good.

status: ready

### CODE REVIEW

**Comment by yenda** [comment]
Trying to solve a bug with react-native https://github.com/facebook/react-native/issues/16458

File: android/app/src/main/java/im/status/ethereum/MainApplication.java [hunk 1]
```diff
@@ -73,7 +74,7 @@ public boolean getUseDeveloperSupport() {
                     new ReactNativeDialogsPackage(),
                     new ImageResizerPackage(),
                     new PickerPackage(),
-                    new WebViewBridgePackage(BuildConfig.DEBUG, callRPC),
```
Comment (rasom) [inline android/app/src/main/java/im/status/ethereum/MainApplication.java]: We definitely need to pass `callRPC` to WebViewBridgePackage

**Comment by oskarth** [comment]
@yenda What's the state of this PR?

File: ios/StatusIm.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -43,6 +43,7 @@
 		81C6E6AE0AA739BE9D87C1D0 /* libPods-StatusImTests.a in Frameworks */ = {isa = PBXBuildFile; fileRef = FC1CBCFE6C906043D6CCEEE1 /* libPods-StatusImTests.a */; };
 		82E689BAF9FB43C8AC6FF1CA /* EvilIcons.ttf in Resources */ = {isa = PBXBuildFile; fileRef = CEB0E2659D1A4F5FA842057A /* EvilIcons.ttf */; };
 		832341BD1AAA6AB300B99B32 /* libRCTText.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 832341B51AAA6A8300B99B32 /* libRCTText.a */; };
+<<<<<<< ours
```
Comment (oskarth) [inline ios/StatusIm.xcodeproj/project.pbxproj]: Merge conflicts shouldn't be committed

**Comment by oskarth** [comment]
Please specify version of RN we are upgrading to; when issue was created latest stable was 0.47, now it is 0.50. Is this the one we use?

**Comment by oskarth** [comment]
Relevant issue for iOS upgrade and unable to open Xcode: https://github.com/facebook/react-native/issues/12261

**Comment by oskarth** [comment]
1. My removal of the merge conflicted pbxproject file was a mistake; please ignore this noop.

2. I thought this RN iOS upgrade stuff would be fairly straightforward with all the work @yenda had already done, but it looks like it is a bunch of indeterminate work. This seems like it is within the scope of https://github.com/status-im/ideas/issues/9 swarm (4 devs committed), and I'd rather spend time kickstarting offline inboxing swarm since I'm leading that one.

3. Why it seems like indeterminate work:
- apply common sense merge (use both and make sure all structures are enclosed properly, remove duplicates) => Xcode won't open.
- Apply only their changes => pod install => whole project file nuked
- Long thread in official repo from February with no canonical solution using current upgrade path

So for all I know it could take a week. It might also take an hour if you find the right combination. One idea is to start RN upgrade from scratch on iOS side, don't use the magic git upgrade script, and then merge that upgrade with this one. We could also put up a bounty to get some Xcode ninjas to resolve this.

**Comment by janherich** [comment]
The last status - except for changes here, one needs to change the import in header files of the `react-native-http-bridge` and `instabug-reactnative` from `#import "RCTBridgeModule.h"` to `#import <React/RCTBridgeModule.h>` (see this thread about the issue - https://github.com/facebook/react-native/issues/15775).
I patched them in my `node_modules` folder, but of course this needs to be fixed in upstream repositories.
After that, the build fails on the linker invocation, which can't find the `GoogleToolboxForMaxLibrary` despite the framework being there and referenced correctly in the `project.pbxproj` file, see screenshot:
screen shot 2017-11-22 at 00 37 55

**Comment by yenda** [comment]
@janherich can you fork the libs until they are fixed upstream because node_modules is local. We might want to do this for all libs anyway

**Comment by oskarth** [comment]
I decided to attack this a bit. I didn't read your comment in detail beforehand, but I did the same thing as you did. Attached patch files which makes it reproducible.

Issue that is quickfix: https://github.com/status-im/status-react/issues/2505

Issue for thing afterwards: https://github.com/status-im/status-react/issues/2506

File: package.json [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "homoglyph-finder": "1.1.1",
     "https-browserify": "0.0.1",
     "identicon.js": "github:status-im/identicon.js",
-    "instabug-reactnative": "git+https://github.com/status-im/instabug-reactnative.git",
```
Comment (oskarth) [inline package.json]: How come you don't update the status-im one? We should use that, not our own forks.

If it is a permission issue and you do this to be unblocked: OK - please ping Carl and/or Nabil and he will sort you (us) out.

Comment (oskarth) [reply package.json]: ^ @janherich

File: package.json [hunk 4]
```diff
@@ -55,7 +55,7 @@
     "react-native-fcm": "10.0.3",
     "react-native-fs": "2.8.1",
     "react-native-http": "github:tradle/react-native-http#834492d",
-    "react-native-http-bridge": "github:yenda/react-native-http-bridge",
+    "react-native-http-bridge": "github:janherich/react-native-http-bridge",
```
Comment (oskarth) [inline package.json]: hot potato :D

File: package.json [hunk 5]
```diff
@@ -7,80 +7,76 @@
...
-    "events": "^1.1.1",
-    "homoglyph-finder": "^1.1.1",
+    "emojilib": "2.2.9",
+    "events": "1.1.1",
+    "homoglyph-finder": "1.1.1",
     "https-browserify": "0.0.1",
     "identicon.js": "github:status-im/identicon.js",
-    "instabug-reactnative": "git+https://github.com/status-im/instabug-reactnative.git",
-    "nfc-react-native": "github:status-im/nfc-react-native",
+    "instabug-reactnative": "git+https://github.com/janherich/instabug-reactnative.git",
```
Comment (oskarth) [inline package.json]: Comment appeared as outdated, not sure why. https://github.com/status-im/status-react/pull/2217/commits/54f5bcbe5bd8bb28ed3157c22cdd3ffce26f8fe7#r152777933

Comment (oskarth) [reply package.json]: I assume this is due to permission issues, ping Carl and Nabil in slack on this

**Comment by janherich** [comment]
@yenda everything works on iOS side (including `re-frisk`/`figwheel` in dev mode) I propose that you will pull the newest changes, make sure that everything is still working on the Android side, and we can then request comprehensive regression testing from QA.
Once this is completed, I can then squash commits, transfer forked repo changes to repositories owned by `status-im` organisation and finally rebase and merge it into develop :)

**Comment by yenda** [comment]
Branch: status-im:update-react-native
Android: https://i.diawi.com/CSM6ZK
iOS: https://i.diawi.com/W9A8hm

**Comment by asemiankevich** [comment]
1. No commands shown after second launch (known, fixed in develop)
2. https://github.com/status-im/status-react/issues/2681. This is fixed in `0.9.10-438-g6948dd17+` branch
3. https://github.com/status-im/status-react/issues/2682. This is fixed in `0.9.10-438-g6948dd17+` branch
4. ios build can not be started
5. https://github.com/status-im/status-react/issues/2697.  This is closed as duplicate of https://github.com/status-im/status-react/issues/2602. Moreover it is in develop and is not related to current branch
6. https://github.com/status-im/status-react/issues/2709

**Comment by asemiankevich** [comment]
### All issues above are fixed in branch below. No new issues found. I assume we can merge this.
----------------
Branch: status-im:update-react-native
Android: https://i.diawi.com/x1acKd
iOS: https://i.diawi.com/bwcauj

**Review by rasom — APPROVED** [review]

**Pull request merged**